### PR TITLE
Fixes #20315 - Sessions mutually exclusive with basic auth

### PR DESCRIPTION
**Pull request by tstrachota**
It fixes http://projects.theforeman.org/issues/20131 too.

### CODE REVIEW

File: lib/hammer_cli_foreman/api/session_authenticator_wrapper.rb [hunk 1]
```diff
@@ -28,13 +29,22 @@ def status
         end
       end
 
+      def force_user_change
+        @user_changed = true
+      end
+
+      def user_changed?
+        @user_changed
```
Comment (ares) [inline lib/hammer_cli_foreman/api/session_authenticator_wrapper.rb]: usually `!!@user_changed` is used to ensure the result of method with `?` at the end returns boolean

**Comment by tstrachota** [comment]
@ares the issues you're mentioning should actually be the main thing this PR is supposed to fix. I couldn't reproduce them. Can you double-check you checked out correct branch and if yes, provide your config files?

**Comment by tstrachota** [comment]
Rebased

**Comment by ares** [comment]
thanks for helping to debug, the `bundle update` helped, now it works as expected, could you just make a note in example foreman.yml that if user enables sessions, stored credentials are ignored? otherwise :+1:

**Comment by tstrachota** [comment]
@ares I added a comment to the default config file and fixed using incorrect password when sessions are off and a different username is passed.

**Comment by ares** [comment]
:+1: works for me

**Pull request merged**